### PR TITLE
feat: key based course translations instead of TARGZ

### PR DESCRIPTION
**Pull request by asadali145**
### What are the relevant tickets?

https://github.com/mitodl/hq/issues/11668

### Description (What does it do?)

Enhances the usage of course translations management command as per the requirements in the issue. translate_course was based on the course TAR, and now it is shifted to the course keys. It will auto export and import courses.

### How can this be tested?

- Create a course in studio to translate
- Go to CMS shell
- Run command `./manage.py cms translate_course --source-course course-v1:TestOrg+Source+Run --target-course course-v1:TestOrg+Arabic+Run --target-language ar --content-translation-provider openai --srt-translation-provider openai --translation-validation-provider openai`
- Replace the source and target keys as required in the command.
- Command should translate the source course and load the translated version in the target course.
- Test different versions like where source and target do not exist, redo translations to test target has updated translations.
- Any other edge cases if you feel if there are any

### CODE REVIEW

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 1]
```diff
@@ -339,23 +360,29 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
...
-            )
+            try:
+                # Create translated copy under the same base directory
+                translated_course_dir = create_translated_copy(
+                    export_dir, target_language
+                )
+            finally:
+                # Remove the export directory once we no longer need it
+                if export_dir.exists():
+                    shutil.rmtree(export_dir)
```
[the remark this answers is not in the document]
Comment (asadali145) [reply src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]: reverted the change

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 2]
```diff
@@ -339,22 +360,25 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
...
+            # find course_dir / "course" without additional discovery logic.
+            export_dir = self._export_source_course(
+                source_course_key, course_translations_base_dir
             )
 
-            # Create translated copy (also under fixed base directory)
-            translated_course_dir = create_translated_copy(
-                extracted_course_dir, target_language
-            )
+            # Create translated copy under the same base directory
```
[the remark this answers is not in the document]
Comment (asadali145) [reply src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py:390]: This results in https://github.com/mitodl/open-edx-plugins/pull/811#discussion_r3420737087

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 3]
```diff
@@ -103,10 +112,23 @@ def add_arguments(self, parser) -> None:
             ),
         )
         parser.add_argument(
-            "--course-dir",
-            dest="course_archive_path",
+            "--source-course",
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]: Let's call it 
```suggestion
            "--source-course-id",
```

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 4]
```diff
@@ -103,10 +112,23 @@ def add_arguments(self, parser) -> None:
...
+            dest="source_course",
+            required=True,
+            help=(
+                "Course key for the source course to export and translate. "
+                "Format: course-v1:ORG+COURSE+RUN"
+            ),
+        )
+        parser.add_argument(
+            "--target-course",
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]: Let's call it 

```suggestion
            "--target-course-id",
```

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 5]
```diff
@@ -241,7 +263,13 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
         """Handle the translate_course command."""
         try:
             start_time = time.perf_counter()
-            course_archive_path = Path(options["course_archive_path"])
+
+            source_course_key = self._parse_course_key(
+                options["source_course"], "--source-course"
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]: nit: We might be able to define them as a constant at the top and use that in both places- the args and here. Same goes for the other arg(target course).

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 6]
```diff
@@ -423,6 +452,147 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
...
+            CommandError: If the key string is invalid
+        """
+        try:
+            return CourseLocator.from_string(key_str)
+        except InvalidKeyError as e:
+            error_msg = (
+                f"Invalid course key for {arg_name}: '{key_str}'. "
+                "Expected format: course-v1:ORG+COURSE+RUN"
+            )
+            raise CommandError(error_msg) from e
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py:508]: I don't think we should add the exception log here since we are already clearly defining in the error message what's wrong here. The people often running this might not want the whole log for a simple self explaining error message.

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py:1]: It might be a good idea to add a confirmation in the command now. Something like:

`You selected the Source course ID <source_course_id>, and the target course ID <target_course_id>. A new course will be created from the source course with the provided target_course_id. Please confirm: y/n.`

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 7]
```diff
@@ -423,6 +452,147 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
...
+            source_course_key: Course key to validate
+
+        Raises:
+            CommandError: If the course does not exist
+        """
+        course = modulestore().get_course(source_course_key)
+        if course is None:
+            error_msg = (
+                f"Source course not found: '{source_course_key}'. "
+                "Please verify the course key and that the course exists."
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]:
```suggestion
                "Please verify that the course key and course exist."
```

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 8]
```diff
@@ -423,6 +452,147 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
...
+            raise CommandError(error_msg)
+
+    def _export_source_course(
+        self, source_course_key: CourseLocator, base_dir: Path
+    ) -> Path:
+        """
+        Export the source course OLX into a temporary directory under base_dir.
+
+        The exported directory will contain a 'course' subdirectory with the full
+        OLX tree, matching the structure that _translate_course_content_async expects.
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]: From Claude: The proper way of referencing methods is like:


`(:meth:`_translate_course_content_async`)`

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
The functionality works well, Suggested a few changes.

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py (hunk 6 above)
Comment (asadali145) [reply src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py:508]: Tried to do this, but raises a pre-commit error. I am sticking with this.

File: src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py [hunk 9]
```diff
@@ -423,6 +460,172 @@ def handle(self, **options) -> None:  # noqa: PLR0915, PLR0912, C901
...
+        self, source_course_id: CourseLocator, target_course_id: CourseLocator
+    ) -> bool:
+        while True:
+            answer = (
+                input(
+                    f"You selected the source course ID {source_course_id}, "
+                    f"and the target course ID {target_course_id}. A new target course "
+                    f"will be created if it does not exist or existing course will be "
+                    f"updated with the translated content if it already exists. "
+                    f"Please confirm: y/n."
```
Comment (arslanashraf7) [inline src/ol_openedx_course_translations/ol_openedx_course_translations/management/commands/translate_course.py]:
```suggestion
                    f"\n\nYou selected the source course ID {source_course_id}, "
                    f"and the target course ID {target_course_id}. A new target course "
                    f"will be created if it does not exist or existing course will be "
                    f"updated with the translated content if it already exists. "
                    f"\n\nPlease confirm: y/n."
```

**Review by arslanashraf7 — APPROVED** [review]
Looks good 👍 Just one suggestion.
